examples/04_manipulating_images/plot_roi_extraction.py

```diff
@@ -1,131 +1,284 @@
+
 fmri_data = fmri_img.get_data()
+# number of voxels being x*y*z, samples in 4th dimension
+print(fmri_data.shape)
+
+##############################################################################
+# The Student’s t-test (:func:`scipy.stats.ttest_ind`) is an established method
+# to determine whether two distributions are statistically different. It can
+# be used to compare voxel time-series from two different experimental
+# conditions (e.g., when houses or faces are shown to individuals during brain
+# scanning). If the time-series distribution is similar in the two conditions,
+# then the voxel is not very interesting to discriminate the condition.
+
+from scipy import stats
+
+# This test returns p-values that represent probabilities that the two
+# time-series had been drawn from the same distribution. The lower is the
+# p-value, the more discriminative is the voxel in distinguishing the two
+# conditions (faces and houses).
 _, p_values = stats.ttest_ind(fmri_data[..., haxby_labels == b'face'],
                               fmri_data[..., haxby_labels == b'house'],
                               axis=-1)
 
 # Use a log scale for p-values
 log_p_values = -np.log10(p_values)
+# NAN values to zero
 log_p_values[np.isnan(log_p_values)] = 0.
 log_p_values[log_p_values > 10.] = 10.
+
+# Visualize statistical p-values using plotting function `plot_stat_map`
 from nilearn.plotting import plot_stat_map
-plot_stat_map(new_img_like(fmri_img, log_p_values),
-              mean_img, title="p-values", cut_coords=cut_coords)
+
+# Before visualizing, we transform the computed p-values to Nifti-like image
+# using function `new_img_like` from nilearn. Otherwise, an error will be
+# raised.
+from nilearn.image import new_img_like
+
+# First argument being a reference image and second argument should be p-values
+# data to convert to a new image as output. This new image will have same header
+# information as reference image.
+log_p_values_img = new_img_like(fmri_img, log_p_values)
+
+# Now, we visualize log p-values image on functional mean image as background
+# with coordinates given manually and colorbar on the right side of plot (by
+# default colorbar=True)
+plot_stat_map(log_p_values_img, mean_img,
+              title="p-values", cut_coords=cut_coords)
 
 ##############################################################################
-# Build a mask from this statistical map
+# Build a mask from this statistical map (Improving the quality of the mask)
+# --------------------------------------------------------------------------
+# **Thresholding** - Voxels with better p-values are kept as voxels of interest.
+# Applying a threshold to an array is easy thanks to numpy indexing à la Matlab.
+# Most simple way of thresholding your data.
 
-# Thresholding
+# Note that we use log p-values data in array and voxels with better p-values
+# lies above value=5 and keeping values less than 5 to zero
```

Comment:
Note that we use log p-values data; we force values below 5 to 0 by thresholding.